### PR TITLE
[RISCV] Collapse fast unaligned access into a single feature [nfc-ish]

### PR DESCRIPTION
**Pull request by preames**
When we'd originally added unaligned-scalar-mem and unaligned-vector-mem, they were separated into two parts under the theory that some processor might implement one, but not the other.  At the moment, we don't have evidence of such a processor.  The C/C++ level interface, and the clang driver command lines have settled on a single unaligned flag which indicates both scalar and vector support unaligned.  Given that, let's remove the test matrix complexity for a set of configurations which don't appear useful.

Given these are internal feature names, I don't think we need to provide any forward compatibility.  Anyone disagree?

Note: The immediate trigger for this patch was finding another case where the unaligned-vector-mem wasn't being properly serialized to IR from clang which resulted in problems reproducing assembly from clang's -emit-llvm feature.  Instead of fixing this, I decided getting rid of the complexity was the better approach.

### CODE REVIEW

File: clang/lib/Driver/ToolChains/Arch/RISCV.cpp [hunk 1]
```diff
@@ -174,15 +174,10 @@ void riscv::getRISCVTargetFeatures(const Driver &D, const llvm::Triple &Triple,
   bool HasV = llvm::is_contained(Features, "+zve32x");
```
Comment (topperc) [inline clang/lib/Driver/ToolChains/Arch/RISCV.cpp:174]: Is HasV unused now?

**Review by topperc — APPROVED** [review]
LGTM

File: clang/lib/Driver/ToolChains/Arch/RISCV.cpp [hunk 2]
```diff
@@ -171,18 +171,12 @@ void riscv::getRISCVTargetFeatures(const Driver &D, const llvm::Triple &Triple,
     Features.push_back("-save-restore");
 
   // -mno-unaligned-access is default, unless -munaligned-access is specified.
-  bool HasV = llvm::is_contained(Features, "+zve32x");
   if (const Arg *A = Args.getLastArg(options::OPT_munaligned_access,
```
Comment (wangpc-pp) [inline clang/lib/Driver/ToolChains/Arch/RISCV.cpp:174]: This whole sippet can be replaced by `tools::AddTargetFeature`.

**Review by wangpc-pp — APPROVED** [review]
LGTM with nit.

**Comment by nico** [comment]
ref https://github.com/rust-lang/rust/pull/118610

**Pull request merged**